src/tests/change.spec.ts

```diff
@@ -0,0 +1,155 @@
+      entity = User
+      fields = ['firstName', 'lastName', 'gender']
+    }
+
+    @Module({
+      imports: [TypeOrmModule.forFeature([User]), TestAuthModule, DefaultCoreModule],
+      exports: [TypeOrmModule],
+    })
+    class RegisteredAdminEntityModule {
+      constructor(private readonly adminSite: DefaultAdminSite) {
+        adminSite.register('user', UserAdmin)
+      }
+    }
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        TestTypeOrmModule.forRoot({ entities: [User, Agency, Group] }),
+        RegisteredAdminEntityModule,
+      ],
+    }).compile()
+
+    const app = module.createNestApplication()
+    await app.init()
+
+    const userData = createTestUser({ firstName: 'Pallavi' })
+    const userRepository: Repository<User> = app.get(getRepositoryToken(User))
+    const user = await userRepository.save(userData)
+    expect(await userRepository.findOneOrFail(user.id)).toBeDefined()
+
+    const server = app.getHttpServer()
+    const req = await request(server).get(`/admin/user/user/${user.id}/change`)
+
+    document.documentElement.innerHTML = req.text
+
+    expect(document.querySelector('label[for="firstName"]')).toBeTruthy()
+    expect(document.querySelector('label[for="lastName"]')).toBeTruthy()
+    expect(document.querySelector('label[for="gender"]')).toBeTruthy()
+
+    expect(document.querySelector('label[for="createdDate"]')).toBeFalsy()
+    expect(document.querySelector('label[for="password"]')).toBeFalsy()
+    expect(document.querySelector('label[for="email"]')).toBeFalsy()
+    expect(document.querySelector('label[for="roles"]')).toBeFalsy()
+    expect(document.querySelector('label[for="moreRoles"]')).toBeFalsy()
+    expect(document.querySelector('label[for="description"]')).toBeFalsy()
+    expect(document.querySelector('label[for="agency"]')).toBeFalsy()
+    expect(document.querySelector('label[for="groups"]')).toBeFalsy()
+    expect(document.querySelector('label[for="age"]')).toBeFalsy()
+    expect(document.querySelector('label[for="numberOfSiblings"]')).toBeFalsy()
+    expect(document.querySelector('label[for="fingerCount"]')).toBeFalsy()
+    expect(document.querySelector('label[for="atomCount"]')).toBeFalsy()
+    expect(document.querySelector('label[for="height"]')).toBeFalsy()
+    expect(document.querySelector('label[for="bmi"]')).toBeFalsy()
+    expect(document.querySelector('label[for="birthdate"]')).toBeFalsy()
+    expect(document.querySelector('label[for="deathdate"]')).toBeFalsy()
+    expect(document.querySelector('label[for="birthtime"]')).toBeFalsy()
+    expect(document.querySelector('label[for="deathtime"]')).toBeFalsy()
+    expect(document.querySelector('label[for="hireDatetime"]')).toBeFalsy()
+    expect(document.querySelector('label[for="isActive"]')).toBeFalsy()
+    expect(document.querySelector('label[for="additionalData"]')).toBeFalsy()
+    expect(document.querySelector('label[for="updatedDate"]')).toBeFalsy()
```

Comment:
Instead of testing that all of these are not displayed, you could check that there's only 3 inputs: `expect(document.querySelector('input')).toHaveLength(3)`